src/ucp/rndv/rndv.c

```diff
@@ -107,7 +107,8 @@ size_t ucp_rndv_rts_pack(ucp_request_t *sreq, ucp_rndv_rts_hdr_t *rndv_rts_hdr,
 
     /* Pack remote keys (which can be empty list) */
     if (UCP_DT_IS_CONTIG(sreq->send.datatype) &&
-        ucp_rndv_is_get_zcopy(sreq, worker->context)) {
+        (UCP_MEM_IS_HOST(sreq->send.mem_type) ||
```

Comment:
will it disable using "put_zcopy" protocol when it is requested by the configuration?

Reply:
No. 
this change i'm trying to differentiate RTS message for HOST and non-HOST when send buffer registration is failed.
for example, HOST buffer reg is fail for read-only pages. CUDA buffer reg can fail because of GDR off.
in this case, I want the receiver to reg the recv buffer if it can for CUDA buffer so that sender can do pipeline to it.
for HOST buffer case,  want to avoid to reg the recv buffer because it can't to put_zcopy for from the sender.
and we dont have a fallback from put_zcopy failure.

Reply:
but for the CPU  memory case, when the receiver gets RTS with valid address and key it will inititate GET-based protocol, even if "put_zcopy" is requested by the config

Reply:
@bureddy why need check host mem type?
i think we can just check sreq->send.state.dt.dt.contig.md_map!=0? or need to add fallback from put to am on sender when getting the RTR?

Reply:
for HOST, sending addr and key so that it will try  get_zcopy at receiver and it will fail. this will avoid registering the receiver buffer (https://github.com/openucx/ucx/blob/master/src/ucp/rndv/rndv.c#L1388).  then the sender will try am_bcopy the after getting RTR w/o key.  otherwise, it will try put_zcopy and fail because sender memory can't register (ex test_ucp_tag_nbx.fallback)

in CUDA case, sending address "0" if md_map = 0, so that it will skip get_zcopy at receiver and registers receive buffer and sender can do pipeline after getting the RTR


Yes, we can simply check est_ucp_tag_nbx.fallback!=0 if we have fallback from put_zcopy to am if sender buffer can;t register